### ISSUE TITLE
redundant calculations in LogLikelihood

### ISSUE THREAD

**Issue by fredRos** [eos/eos#6]
Currently, the transversity amplitudes are calculated for each observable separately ,although once would be enough. 

Computing the integrated amplitudes is the expensive part. Observables should be calculated only from those intermediate results. We need to refactor such that observables get those results without computing them themselves.

**Comment by dvandyk** [comment]
- [ ] Introduce new abstract base class `CacheableObservable`
- [ ] Add implementation class template `ConcreteCacheableObservable<Type_, Cache_>` and factory:
  - factory should follow the signature:
``` c++
template <typename T_, typename Args_ ...>
make_cacheable_observable(
      "NAME",
      &T_::observable_from_cache,
      &T_::cache_from_kinematics,
      std::tuple<Args_ ...> kinematics
)
```
  - in this step, no caching is done yet. We simply separate the calculation of the intermedate result (to be stored in the cache) from the final result for the observable
  - `ConcreateCacheableObservable` should store the most recently generated instance of its cache.
- [ ] Add implemenation class template `AliasedCacheableObservable<Type_, Cache_>`, which is only to be constructed from within `ObservableCache'
- [ ] In `ObservableCache`, when comparing a newly added `shared_ptr<Observable>` with the existing one, replace the observable with an aliased observable.

**Comment by dvandyk** [comment]
Resolved by PR #280
